edgelet/iotedge/src/check.rs

```diff
@@ -0,0 +1,749 @@
+                    ),
+                ],
+            ),
+            (
+                "config",
+                &[
+                    ("config.yaml has correct hostname", settings_hostname),
+                    (
+                        "config.yaml has well-formed moby runtime URI",
+                        settings_moby_runtime_uri,
+                    ),
+                    (
+                        "config.yaml has correct URIs for daemon mgmt endpoint",
+                        daemon_mgmt_endpoint_uri,
+                    ),
+                    (
+                        "container runtime network allows name resolution",
+                        container_runtime_network,
+                    ),
+                ],
+            ),
+            (
+                "deps",
+                &[
+                    ("latest security daemon", iotedged_version),
+                    ("latest edge agent container image", |check| {
+                        edge_container_version(
+                            "edgeAgent",
+                            &check.latest_versions.edge_agent,
+                            check.docker_host_arg.as_ref().map(AsRef::as_ref),
+                        )
+                    }),
+                    ("latest edge hub container image", |check| {
+                        edge_container_version(
+                            "edgeHub",
+                            &check.latest_versions.edge_hub,
+                            check.docker_host_arg.as_ref().map(AsRef::as_ref),
+                        )
+                    }),
+                    ("host time is close to real time", host_local_time),
+                    ("container time is close to host time", container_local_time),
+                ],
+            ),
+            (
+                "conn",
+                &[
+                    ("can connect to IoT Hub AMQP port", |check| {
+                        connection_to_iot_hub(check, 5671)
+                    }),
+                    ("can connect to IoT Hub HTTPS port", |check| {
+                        connection_to_iot_hub(check, 443)
+                    }),
+                    ("can connect to IoT Hub MQTT port", |check| {
+                        connection_to_iot_hub(check, 8883)
+                    }),
+                ],
+            ),
+        ];
+
+        let mut steps_override: HashSet<Cow<'static, str>> =
+            if let Some(steps_override) = self.steps_override.take() {
+                steps_override.into_iter().map(Into::into).collect()
+            } else {
+                CHECKS
+                    .iter()
+                    .map(|&(section_name, _)| section_name.into())
+                    .collect()
+            };
+        steps_override.insert(PRE_CHECK_SECTION_NAME.into());
+
+        let mut have_warnings = false;
+        let mut have_errors = false;
+
+        for (section_name, section_checks) in CHECKS {
```

Comment:
Perhaps we should break out the printing of the results so that we can add some tests?

Reply:
I've added some unit tests. The tests run the individual check functions themselves, so this `execute_inner` itself doesn't need to change.